packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { isUndefined } from 'lodash';
+import React from 'react';
+import { isEmpty } from '../../utils';
+import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
+import { CheckboxHintText } from './CheckboxHintText';
+import { CheckboxIcon } from './CheckboxIcon';
+import { CheckboxInput } from './CheckboxInput';
+import { CheckboxLabel } from './CheckboxLabel';
+import { CheckboxLabelText } from './CheckboxLabelText';
+import { useCheckbox } from './useCheckbox';
+import { Wrapper } from './Wrapper';
+
+type CheckboxProps = {
+  /**
+   * If `true`, sets the checkbox will be checked.
+   * Use `onChange` to update its value
+   *
+   * @default false
+   */
+  isChecked?: boolean;
+  /**
+   * If `true`, the checkbox will be initially checked.
+   *
+   * @default false
+   */
```

Comment:
can we somehow add a lint rule or TS rule to enforce these comments for certain directories? I feel this is a nice practice and gives better description in intellisense

Reply:
Ahh yeah no don't think we have a eslint rule. We just have to make it a habit for public facing APIs to provide good inline descriptions.